### PR TITLE
Add SELinux Support for CRI

### PR DESCRIPTION
**Pull request by crosbymichael**
This is a carry of #1246.  

We are splitting up the work and I will be carrying this PR to merge.

### CODE REVIEW

**Comment by crosbymichael** [comment]
I went ahead and squashed this as it had many WIP commits and unsigned commits from the previous PR causing CI to fail.

File: hack/utils.sh [hunk 1]
```diff
@@ -17,9 +17,9 @@
 ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"/..
 
 # Not from vendor.conf.
-CRITOOL_VERSION=v1.18.0
+CRITOOL_VERSION=selinux
 CRITOOL_PKG=github.com/kubernetes-sigs/cri-tools
-CRITOOL_REPO=github.com/kubernetes-sigs/cri-tools
+CRITOOL_REPO=github.com/erikwilson/cri-tools
```
Comment (crosbymichael) [inline hack/utils.sh:22]: Just making a comment as a note, we will have to update this back to upstream once all the upstream PRs are merged into cri-tools

**Comment by mikebrow** [comment]
e2e bucket timed out for no obvious reason restarting

**Comment by crosbymichael** [comment]
I'm not getting good info on why this is timing out....

File: hack/utils.sh (hunk 1 above)
Comment (erikwilson) [reply hack/utils.sh:22]: The initial PR for MCS testing has been merged and is now available at https://github.com/kubernetes-sigs/cri-tools/commit/89384cc13a27bb9128553c9fe75a7cc07c6a95bb

Comment (crosbymichael) [reply hack/utils.sh:22]: Thanks, i'll update back to upstream

**Comment by crosbymichael** [comment]
/test pull-cri-containerd-node-e2e

it looks like a normal timeout now for a single test, going to rerun

**Comment by crosbymichael** [comment]
victory!

**Comment by crosbymichael** [comment]
Tested labeling for volumes:

```bash
system_u:system_r:container_t:s0:c192,c450 root 9226 0.0  0.1 10636 5168 ?       Ss   11:57   0:00 nginx: master process nginx -g daemon off;
system_u:system_r:container_t:s0:c192,c450 101 9240 0.0  0.0 11100 2552 ?        S    11:57   0:00 nginx: worker process
```

process ^^ 

```bash
drwxr-xr-x. 2 root root unconfined_u:object_r:mnt_t:s0                  6 May 26 11:57 mnt
drwxr-xr-x. 2 root root system_u:object_r:container_file_t:s0:c192,c450 6 May 26 11:57 test
```

mount ^^

**Comment by crosbymichael** [comment]
I think we could merge this and then follow up on adding reference counting to the pod level and off of the container.

File: pkg/server/sandbox_run.go [hunk 2]
```diff
@@ -157,6 +158,18 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		return nil, errors.Wrap(err, "failed to generate sandbox container spec")
 	}
 	log.G(ctx).Debugf("Sandbox container %q spec: %#+v", id, spew.NewFormatter(spec))
+	sandbox.ProcessLabel = spec.Process.SelinuxLabel
+	defer func() {
+		if retErr != nil {
+			_ = label.ReleaseLabel(sandbox.ProcessLabel)
```
Comment (mxpv) [inline pkg/server/sandbox_run.go:164]: nit: this one is deprecated. We can use `selinux.ReleaseLabel` which returns nothing, so we don't have to write `_ = X` everywhere.

Comment (crosbymichael) [reply pkg/server/sandbox_run.go:164]: I'll add this for the next around of updates.

Comment (crosbymichael) [reply pkg/server/sandbox_run.go]: i went ahead and updated this

**Comment by crosbymichael** [comment]
Updated, please take another look

File: pkg/server/service.go [hunk 3]
```diff
@@ -99,12 +100,13 @@ type criService struct {
...
 func NewCRIService(config criconfig.Config, client *containerd.Client) (CRIService, error) {
 	var err error
+	labels := label.NewStore()
 	c := &criService{
 		config:             config,
 		client:             client,
 		os:                 osinterface.RealOS{},
-		sandboxStore:       sandboxstore.NewStore(),
-		containerStore:     containerstore.NewStore(),
+		sandboxStore:       sandboxstore.NewStore(labels),
```
Comment (dmcgowan) [inline pkg/server/service.go:108]: We should try to decouple this a little bit, maybe having the label store could be registered through the plugin system in the future.

Comment (dmcgowan) [reply pkg/server/service.go:108]: In a follow up or maybe post merge back into containerd

**Comment by crosbymichael** [comment]
Ya, the store is what I’m currently working on and will be putting a PR up shortly.

Michael Crosby

**Pull request merged**